lib/fastlane/plugin/appcenter/actions/appcenter_upload_action.rb

```diff
@@ -131,15 +143,18 @@ def self.run_release_upload(params)
 
         file_ext = Helper::AppcenterHelper.file_extname_full(file)
         if destination_type == "group"
-          UI.user_error!("Can't distribute #{file_ext} to groups, please use `destination_type: 'store'`") if %w(.aab).include? file_ext
+          UI.user_error!("Can't distribute #{file_ext} to groups, please use `destination_type: 'store'`") if Constants::STORE_ONLY_EXTENSIONS.include? file_ext
         else
-          UI.user_error!("Can't distribute #{file_ext} to stores, please use `destination_type: 'group'`") if %w(.app .app.zip .dmg .pkg).include? file_ext
+          UI.user_error!("Can't distribute #{file_ext} to stores, please use `destination_type: 'group'`") unless Constants::STORE_SUPPORTED_EXTENSIONS.include? file_ext
         end
 
         unless params[:file].to_s.empty?
-          if %w[.dmg .pkg].include? file_ext
+          if Constants::FULL_VERSION_REQUIRED_EXTENSIONS.include? file_ext
             UI.user_error!("Fields `version` and `build_number` must be specified to upload a #{file_ext} file") if build_number.to_s.empty? || version.to_s.empty?
             release_upload_body = { build_version: version, build_number: build_number }
+          elsif Constants::VERSION_REQUIRED_EXTENSIONS.include? file_ext
```

Comment:
file_ext will probably contain ".zip" for ".app.zip" files. This requirement of having a version number should be OS dependent as well.

Reply:
Good catch!